tsdb/store.go

```diff
@@ -1485,6 +1511,15 @@ func (s *Store) DeleteSeries(database string, sources []influxql.Source, conditi
 		max = influxql.MaxTime
 	}
 
+	shardFilter := make([]func(sh *Shard) bool, 0)
+	shardFilter = append(shardFilter, byDatabase(database))
+
+	if len(sources) != 0 {
+		if rpName := sources[0].(*influxql.Measurement).RetentionPolicy; rpName != "" {
+			shardFilter = append(shardFilter, byRetentionPolicy(rpName))
+		}
+	}
+
```

Comment:
If a retention policy is specified, it seems like we want to filter on both the retention policy _and_ the measurement, not retention policy _or_ measurement. This feels like it will delete all measurements on a retention policy. 

If this is true, I think that `filterShards` can be reverted to only taking a single function.

Reply:
Or at the very least do extra work by including shards that don't have the target measurement, just the retention policy matching.

Reply:
Even if the `OR` logic is correct (which I don't think it is), this could be combined into one function, simplifying `filterShards` and making the overall logic easier to understand.